domain/domain.go

```diff
@@ -433,52 +436,72 @@ func (m *MockFailure) getValue() bool {
 }
 
 type schemaValidityInfo struct {
-	isValid          bool
-	firstValidTS     uint64 // It's used for recording the first txn TS of schema vaild.
-	mux              sync.RWMutex
-	lastReloadTime   int64       // It's used for recording the time of last reload schema.
-	lastSuccTS       uint64      // It's used for recording the last txn TS of loading schema succeed.
+	mux           sync.RWMutex
+	isExpired     bool   // Whether information schema is out of date.
+	recoveredTS   uint64 // It's used for recording the first txn TS of schema vaild.
+	lastSchemaVer int64  // It's used for recording the last schema version.
+	timeInfo      struct {
+		mux            sync.RWMutex
+		lastReloadTime int64  // It's used for recording the time of last reload schema.
+		lastSuccTS     uint64 // It's used for recording the last txn TS of loading schema succeed.
+	}
 	MockReloadFailed MockFailure // It mocks reload failed.
 }
 
+func (s *schemaValidityInfo) updateSchemaVersion(version int64) {
+	atomic.StoreInt64(&s.lastSchemaVer, version)
```

Comment:
I don't suggest use both atomic and RWMutex...
schemaValidityInfo is already a mess, it makes things from bad to worse.
If any bug caused by this style, it will be the most difficult bug to debug.